File: Models/EngagementPoints.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace GitlabInfo.Models
{
    public class EngagementPoints
    {
        public User AwardingUser { get; set; }
        public User ReceivingUser { get; set; }
        public int Points { get; set; }
        public DateTime ReceivingDate { get; set; }
    }
}

File: ClientApp/src/app/project-management/models/engagementPoints.ts
import { IUser } from '../../models/user';

export interface IEngagementPointsPut {
    projectId: number;
    receivingUser: IUser;
    points: number;
    bonus: boolean;
    comment: string;
}

export interface IEngagementPointsGet {
    awardingUser: IUser;
    receivingUser: IUser;
    points: number;
    receivingDate: Date;
}
